### DISCUSSION TITLE
Could the overlay consider integrating runtime errors as well?

### DISCUSSION THREAD

**Discussion by nanianlisao** [web-infra-dev/rsbuild#2126] [Ideas]
The current overlay only handles compilation errors. Have you considered integrating runtime errors as well?

Of course, from an experience standpoint, I believe it would be better to directly display the source code after processing the error stack, similar to what react-error-overlay does.

Unfortunately, react-error-overlay has not been maintained for a long time, which has led to some unresolved issues.

**Comment by 9aoy** [comment]
@nanianlisao  Thanks for your advice. Yes, from a developer's perspective, I would like to be able to notice a runtime error as soon as possible. 
But I'm worried about whether the current nearly full-screen UI of rsbuild overlay is suitable for runtime error display, maybe it should be less intrusive for runtime errors. Actually, a runtime error overlay can be used in various build tools and frameworks, so I prefer it to be an external package.

**Reply by nanianlisao** [reply]
I believe it could be configured like webpack-dev-server, to allow for some filtering of runtime errors.

I agree with your view on acting as an external package, similar to react-error-overlay, but I hope to use a common UI.

Could you share how you currently handle runtime errors? If not, do you have any plans to implement this?
